activation/poetdb.go

```diff
@@ -0,0 +1,135 @@
+package activation
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	xdr "github.com/nullstyle/go-xdr/xdr3"
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/database"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/types"
+	"github.com/spacemeshos/merkle-tree"
+	"github.com/spacemeshos/poet/hash"
+	"github.com/spacemeshos/poet/shared"
+	"github.com/spacemeshos/poet/verifier"
+	"github.com/spacemeshos/sha256-simd"
+)
+
+type PoetDb struct {
+	store database.Database
+	log   log.Log
+}
+
+func NewPoetDb(store database.Database, log log.Log) *PoetDb {
+	return &PoetDb{store: store, log: log}
+}
+
+func (db *PoetDb) ValidateAndStorePoetProof(proof types.PoetProof, poetId []byte, roundId uint64, signature []byte) error {
+	root, err := calcRoot(proof.Members)
+	if err != nil {
+		return db.logError(fmt.Errorf("failed to calculate membership root for poetId %x round %d: %v",
```

Comment:
please use bytes to hex when printing byte arrays

Reply:
instead of `%x`? why?

Reply:
https://play.golang.org/p/01DApi_ig3L